lib/template-parts.php

```diff
@@ -212,12 +212,20 @@ function filter_rest_wp_template_part_query( $args, $request ) {
 		// Ensure auto-drafts of all theme supplied template parts are created.
 		if ( wp_get_theme()->get( 'TextDomain' ) === $request['theme'] ) {
 			// Get file paths for all theme supplied template parts.
-			$template_part_files = glob( get_stylesheet_directory() . '/block-template-parts/*.html' );
-			$template_part_files = is_array( $template_part_files ) ? $template_part_files : array();
+			function get_template_part_paths( $base_directory, $path_list = array() ) {
+				if ( file_exists( $base_directory . '/block-template-parts' ) ) {
+					$nested_files      = new RecursiveIteratorIterator( new RecursiveDirectoryIterator( $base_directory . '/block-template-parts' ) );
+					$nested_html_files = new RegexIterator( $nested_files, '/^.+\.html$/i', RecursiveRegexIterator::GET_MATCH );
+					foreach ( $nested_html_files as $path => $file ) {
+						$path_list[] = $path;
+					}
+				}
+				return $path_list;
+			}
+
+			$template_part_files = get_template_part_paths( get_stylesheet_directory() );
 			if ( is_child_theme() ) {
-				$child_template_part_files = glob( get_template_directory() . '/block-template-parts/*.html' );
-				$child_template_part_files = is_array( $child_template_part_files ) ? $child_template_part_files : array();
-				$template_part_files       = array_merge( $template_part_files, $child_template_part_files );
+				$template_part_files = get_template_part_paths( get_template_directory(), $template_part_files );
```

Comment:
Could we concatenate the arrays outside of the function instead of passing it in to modify? I like how that would make the function "more pure," if that makes sense

Reply:
Yeah, that makes sense.  As in b86514f?